### PR TITLE
Feature/cuda const

### PR DESCRIPTION
**Pull request by nanzifan**
Add CUDA constant memory support memory allocator named DEVICE_CONST.
Add one example that use constant memory allocator.

### CODE REVIEW

File: src/umpire/ResourceManager.cpp [hunk 1]
```diff
@@ -150,6 +162,16 @@ ResourceManager::getAllocator(resource::MemoryResourceType resource_type)
   return Allocator(m_memory_resources[resource_type]);
 }
 
+void
+ResourceManager::registerAllocator(const std::string& name, Allocator allocator)
```
Comment (davidbeckingsale) [inline src/umpire/ResourceManager.cpp]: Why have you moved this method?

Comment (davidbeckingsale) [reply src/umpire/ResourceManager.cpp]: Okay - could you fix this please?

Comment (nanzifan) [reply src/umpire/ResourceManager.cpp]: No problem.

Comment (davidbeckingsale) [reply src/umpire/ResourceManager.cpp]: Thanks :)

Comment (nanzifan) [reply src/umpire/ResourceManager.cpp]: I put them back to original place. ^.^

**Pull request merged**